spack install gcc@5.4.0 fails

As the title says, on my machine spack fails when compiling gcc, when following the  [instructions](https://picongpu.readthedocs.io/en/0.4.0-rc2/install/instructions/spack.html).

[error log](https://gist.github.com/berceanu/491ee214eb5e3131ba11b0762e7efe00) (large file!)
[system info](https://gist.github.com/berceanu/24a8b568f618cae4af4889a9e2ffc3c3)

@n01r , hard to say for sure -- macos often has problems I'm too lazy to explore. The error is raised when i do `spack install gcc@5.4.0` and it looks like this
```
==> Installing gcc
==> Using cached archive: /Users/igor/src/spack/var/spack/cache/gcc/gcc-5.4.0.tar.bz2
==> Staging archive: /Users/igor/src/spack/var/spack/stage/gcc-5.4.0-hrisaxyfs27fmku7aaxha6vamdu6kuyz/gcc-5.4.0.tar.bz2
==> Created stage in /Users/igor/src/spack/var/spack/stage/gcc-5.4.0-hrisaxyfs27fmku7aaxha6vamdu6kuyz
==> Applied patch darwin/gcc-7.1.0-headerpad.patch
==> Applied patch darwin/gcc-6.1.0-jit.patch
==> Applied patch ucontext_t.patch
==> Applied patch stack_t.patch
==> Applied patch signal.patch
==> Ran patch() for gcc
==> Building gcc [AutotoolsPackage]
==> Executing phase: 'autoreconf'
==> Executing phase: 'configure'
==> Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    'make' '-j8' 'bootstrap'

5 errors found in build log:
     9884     /Users/igor/src/spack/var/spack/stage/gcc-5.4.0-hrisaxyfs27fmku7a
              axha6vamdu6kuyz/gcc-5.4.0/gcc/wide-int.h:284:18: note: did you me
              an class here?
     9885     template <int N> struct fixed_wide_int_storage;
     9886                      ^~~~~~
     9887                      class
     9888     In file included from /Users/igor/src/spack/var/spack/stage/gcc-5
              .4.0-hrisaxyfs27fmku7aaxha6vamdu6kuyz/gcc-5.4.0/gcc/auto-profile.
              c:25:
     9889     In file included from /Applications/Xcode.app/Contents/Developer/
              Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:446:
  >> 9890     /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefaul
              t.xctoolchain/usr/include/c++/v1/functional:1408:2: error: no mem
              ber named 'fancy_abort' in namespace 'std::__1'; did you mean sim
              ply 'fancy_abort'?
     9891             _VSTD::abort();
     9892             ^~~~~~~
     9893     /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefaul
              t.xctoolchain/usr/include/c++/v1/__config:441:15: note: expanded 
              from macro '_VSTD'
     9894     #define _VSTD std::_LIBCPP_NAMESPACE
     9895                   ^
     9896     /Users/igor/src/spack/var/spack/stage/gcc-5.4.0-hrisaxyfs27fmku7a
              axha6vamdu6kuyz/gcc-5.4.0/gcc/system.h:700:13: note: 'fancy_abort
              ' declared here

     ...

     10255    ^
     10256    /Users/igor/src/spack/var/spack/stage/gcc-5.4.0-hrisaxyfs27fmku7a
              axha6vamdu6kuyz/gcc-5.4.0/gcc/coretypes.h:55:8: note: previous us
              e is here
     10257    struct rtx_def;
     10258           ^
     10259    13 warnings and 1 error generated.
     10260    clang: warning: treating 'c' input as 'c++' when in C++ mode, thi
              s behavior is deprecated [-Wdeprecated]
  >> 10261    make[3]: *** [auto-profile.o] Error 1
     10262    make[3]: *** Waiting for unfinished jobs....
     10263    In file included from /Users/igor/src/spack/var/spack/stage/gcc-5
              .4.0-hrisaxyfs27fmku7aaxha6vamdu6kuyz/gcc-5.4.0/gcc/bb-reorder.c:
              95:
     10264    /Users/igor/src/spack/var/spack/stage/gcc-5.4.0-hrisaxyfs27fmku7a
              axha6vamdu6kuyz/gcc-5.4.0/gcc/rtl.h:282:1: warning: 'rtx_def' def
              ined as a struct here but previously declared as a class [-Wmisma
              tched-tags]
     10265    struct GTY((desc("0"), tag("0"),
     10266    ^
     10267    /Users/igor/src/spack/var/spack/stage/gcc-5.4.0-hrisaxyfs27fmku7a
              axha6vamdu6kuyz/gcc-5.4.0/gcc/coretypes.h:62:1: note: did you mea
              n struct here?

     ...

     11252    echo timestamp > s-automata
     11253    /bin/sh /Users/igor/src/spack/var/spack/stage/gcc-5.4.0-hrisaxyfs
              27fmku7aaxha6vamdu6kuyz/gcc-5.4.0/gcc/../move-if-change tmp-attrt
              ab.c    insn-attrtab.c
     11254    /bin/sh /Users/igor/src/spack/var/spack/stage/gcc-5.4.0-hrisaxyfs
              27fmku7aaxha6vamdu6kuyz/gcc-5.4.0/gcc/../move-if-change tmp-dfata
              b.c     insn-dfatab.c
     11255    /bin/sh /Users/igor/src/spack/var/spack/stage/gcc-5.4.0-hrisaxyfs
              27fmku7aaxha6vamdu6kuyz/gcc-5.4.0/gcc/../move-if-change tmp-laten
              cytab.c insn-latencytab.c
     11256    echo timestamp > s-attrtab
     11257    rm gcc.pod
  >> 11258    make[2]: *** [all-stage1-gcc] Error 2
  >> 11259    make[1]: *** [stage1-bubble] Error 2
  >> 11260    make: *** [bootstrap] Error 2

See build log for details:
  /Users/igor/src/spack/var/spack/stage/gcc-5.4.0-hrisaxyfs27fmku7aaxha6vamdu6kuyz/gcc-5.4.0/spack-build.out
```

--- a/docs/source/install/instructions/spack.rst
+++ b/docs/source/install/instructions/spack.rst
@@ -31,7 +31,7 @@ First `install spack <http://spack.readthedocs.io/en/latest/getting_started.html
    spack bootstrap
 
    # install a supported compiler
-   spack compiler list | grep gcc@5.4.0 | spack install gcc@5.4.0 && spack load gcc@5.4.0 && spack compiler add
+   spack compiler list | grep gcc@7.3.0 | spack install gcc@7.3.0 && spack load gcc@7.3.0 && spack compiler add
 
    # add the PIConGPU repository
    git clone https://github.com/ComputationalRadiationPhysics/spack-repo.git $HOME/src/spack-repo
@@ -52,7 +52,7 @@ The installation of the latest version of PIConGPU is now as easy as:
 
 .. code-block:: bash
 
-   spack install picongpu %gcc@5.4.0
+   spack install picongpu %gcc@7.3.0
 
 Use PIConGPU
 ^^^^^^^^^^^^
@@ -74,3 +74,22 @@ For more information on *variants* of the ``picongpu`` package in spack run ``sp
 
       spack install picongpu backend=omp2b
       spack load picongpu backend=omp2b
+
+.. note::
+
+   If the install fails or you want to compile for CUDA 8.0, try using GCC 5.3.0:
+
+   .. code-block:: bash
+
+      spack compiler list | grep gcc@5.3.0 | spack install gcc@5.3.0 && spack load gcc@5.3.0 && spack compiler add
+      spack install picongpu %gcc@5.3.0
+      spack load picongpu %gcc@5.3.0
+
+   If the install fails or you want to compile for CUDA 9.0/9.1, try using GCC 5.5.0:
+
+   .. code-block:: bash
+
+      spack compiler list | grep gcc@5.5.0 | spack install gcc@5.5.0 && spack load gcc@5.5.0 && spack compiler add
+      spack install picongpu %gcc@5.5.0
+      spack load picongpu %gcc@5.5.0
+

